api/api.go

```diff
@@ -0,0 +1,149 @@
+		http.ServeFile(w, r, "static/index.html")
+	})
+
+	// Define the API router
+	apiRouter := r.PathPrefix("/api").Subrouter()
+
+	// Public Endpoints
+	apiRouter.HandleFunc("/signup", makeHTTPHandleFunc(s.handleCreateAccount)).Methods("POST")
+	apiRouter.HandleFunc("/signup", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "static/signup.html")
+	}).Methods("GET")
+	apiRouter.HandleFunc("/login", makeHTTPHandleFunc(s.handleLogin)).Methods("POST")
+	apiRouter.HandleFunc("/login", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "static/login.html")
+	}).Methods("GET")
+	apiRouter.HandleFunc("/logout", makeHTTPHandleFunc(s.handleLogout)).Methods("POST")
+	apiRouter.HandleFunc("/events", makeHTTPHandleFunc(s.handleGetEvents)).Methods("GET")
+	apiRouter.HandleFunc("/events/{eventId}/entries", makeHTTPHandleFunc(s.handleGetEventEntries)).Methods("GET")
+	apiRouter.HandleFunc("/memberships", makeHTTPHandleFunc(s.handleGetMemberships)).Methods("GET")
+
+	// TODO refresh token
+
+	// Authenticated User Endpoints with JWT authentication
+	authRouter := apiRouter.PathPrefix("/").Subrouter()
+	authRouter.Use(withJWTAuth)
+
+	authRouter.HandleFunc("/account", makeHTTPHandleFunc(s.handleGetAccount)).Methods("GET")
+	authRouter.HandleFunc("/account", makeHTTPHandleFunc(s.handleUpdateAccount)).Methods("PUT")
+	authRouter.HandleFunc("/account", makeHTTPHandleFunc(s.handleDeleteAccount)).Methods("DELETE")
+	authRouter.HandleFunc("/account/memberships", makeHTTPHandleFunc(s.handleGetAccountMemberships)).Methods("GET")
+	authRouter.HandleFunc("/account/events", makeHTTPHandleFunc(s.handleGetAccountEvents)).Methods("GET")
+	authRouter.HandleFunc("/memberships/{membershipId}/purchase", makeHTTPHandleFunc(s.handleCreateAccountMembership)).Methods("POST")
+	authRouter.HandleFunc("/events/{eventId}/entries", makeHTTPHandleFunc(s.handleCreateEntry)).Methods("POST")
+	authRouter.HandleFunc("/entries/{entryId}", makeHTTPHandleFunc(s.handleDeleteEntry)).Methods("DELETE")
+
+	// Admin Endpoints with JWT authentication
+	adminRouter := apiRouter.PathPrefix("/admin").Subrouter()
+	adminRouter.Use(withJWTAuth)
+	adminRouter.Use(s.isAdmin)
+
+	adminRouter.HandleFunc("/events", makeHTTPHandleFunc(s.handleCreateEvent)).Methods("POST")
+	adminRouter.HandleFunc("/events/{eventId}", makeHTTPHandleFunc(s.handleDeleteEvent)).Methods("DELETE")
+	adminRouter.HandleFunc("/memberships", makeHTTPHandleFunc(s.handleCreateMembership)).Methods("POST")
+	adminRouter.HandleFunc("/memberships/{membershipId}", makeHTTPHandleFunc(s.handleDeleteMembership)).Methods("DELETE")
+
+	log.Println("JSON API server running on port: ", s.listenAddr)
+
+	if err := http.ListenAndServe(s.listenAddr, r); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func makeHTTPHandleFunc(f apiFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err := f(w, r); err != nil {
+			log.Println("nested function failed")
+			writeErrorJSON(w, err)
+		}
+	}
+}
```

Comment:
This is weird. You wrapped every handler into this error handling "middleware". Although you may have saved some code it is overly generic. I would let the handlers handle this behaviour themselves. 
Image someone contributing to your project and he forgets to wrap it in the router or he handles the error already in the handler and it gets double wrapped. Just seems too magical to me. Also, you had to wrap every handler in this function and that should already smell.